### PR TITLE
User story2

### PR DESCRIPTION
**Pull request by reyk-zepper**
- create EditForm component
- add Immer and useImmer
- add Zustand
- add edit function
- add delte function

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 This is a [Next.js](https://nextjs.org/) project bootstrapped with [`create-next-app`](https://github.com/vercel/next.js/tree/canary/packages/create-next-app).
 
-## Getting Started
+## Get ting Started
```
Comment (lenzloesche) [inline README.md:3]: why did you change this? :)

File: components/EditForm/index.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+  return (
+    <form onSubmit={handleSubmit}>
+      <input
+        type="text"
+        name="projectname"
+        placeholder="project name"
+        value={projectName}
+        onChange={handleProjectNameChange}
+      />
```
Comment (lenzloesche) [inline components/EditForm/index.js:49]: `required` might be useful for the project name

File: components/Form/index.js [hunk 3]
```diff
@@ -37,68 +26,68 @@ export default function Form() {
...
     };
 
     const resetItems = partList.map((item) => {
       return { ...item, price: 0, value: "" };
     });
 
-    //das neue Projekt zur Liste der Produkte hinzufügen
-    setProjects([...projects, project]);
+    //add new project to list
+    addProject(project);
```
Comment (lenzloesche) [inline components/Form/index.js:41]: usually the comments should be sparse. in his case the code already says: addProject, so it is clear what is done here.

File: components/Form/index.js [hunk 4]
```diff
@@ -37,68 +26,68 @@ export default function Form() {
...
-              name={item.name}
-              type="text"
-              placeholder={item.name}
-              value={item.value}
-              onChange={(event) =>
-                handleItemChange(index, "value", event.target.value)
-              }
-            />
+    <>
+      <form onSubmit={handleSubmit}>
```
Comment (lenzloesche) [inline components/Form/index.js:52]: it seems that you have a similar form in `EditForm` already. Maybe make a component out of the form? Might be a possible solution.

File: components/ProjectCards/ProjectCard.test.js [hunk 5]
```diff
@@ -1,29 +1,33 @@
...
+    expect(productCardsComponent).toBeNull();
   });
 
-  it("should calculate the total price correctly", () => {
-    const { getByText } = render(<ProjectCard projects={projects} />);
+  it("should pass correct props to ProductCard components", () => {
+    const { queryAllByTestId } = render(
+      <ProjectCards projects={mockProjects} />
+    );
+    const productCardComponents = queryAllByTestId("product-card");
```
Comment (lenzloesche) [inline components/ProjectCards/ProjectCard.test.js:21]: I think that we aren't supposed to use test ids, maybe we ask Martin on Monday.

**Review by lenzloesche — COMMENTED** [review]
great work!
I have some suggestions.

File: README.md (hunk 1 above)
Comment (reyk-zepper) [reply README.md:3]: you can't commit a brand new branch without changes, that"s for the intial commit of the branch.

File: components/EditForm/index.js (hunk 2 above)
Comment (reyk-zepper) [reply components/EditForm/index.js:49]: i will add verification in an other user story

File: components/Form/index.js (hunk 3 above)
Comment (reyk-zepper) [reply components/Form/index.js:41]: this comment is also in german in an english code...so it was a notice for me to learn, but thx for remember🦄

File: components/Form/index.js (hunk 4 above)
Comment (reyk-zepper) [reply components/Form/index.js:52]: ...that was my idea too, but that are similiar forms they do different things...so instead of prop drilling i decide to duplicate -> duplication is cheaper than a wrong abstraction
...i will keep this in mind, if i need this form again
